src/test/java/com/mathworks/ci/systemtests/TestResultVisualizationIT.java

```diff
@@ -0,0 +1,340 @@
+
+        String hyperlinkInSummary = getHyperlinkFromBuildStatus(build);
+
+        String hyperlinkOfTestResultsTab = getTestResultTabLinkFromSidePanel(build);
+
+        assertTrue(hyperlinkOfTestResultsTab.contains(hyperlinkInSummary));
+    }
+
+    @Test
+    public void verifyContentInTestResultsTable() throws Exception{
+        FreeStyleProject project = jenkins.createFreeStyleProject();
+        project.setScm(new ExtractResourceSCM(Utilities.getURLForTestData()));
+
+        UseMatlabVersionBuildWrapper buildWrapper = new UseMatlabVersionBuildWrapper();
+        buildWrapper.setMatlabBuildWrapperContent(new MatlabBuildWrapperContent(Message.getValue("matlab.custom.location"), Utilities.getMatlabRoot()));
+        project.getBuildWrappersList().add(buildWrapper);
+
+        // Run tests through Run Build step
+        RunMatlabBuildBuilder buildtoolBuilder = new RunMatlabBuildBuilder();
+        buildtoolBuilder.setTasks("test");
+        project.getBuildersList().add(buildtoolBuilder);
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        // Verify button text
+        String totalTestsButtonText = getButtonText("TOTAL", build);
+        assertTrue(totalTestsButtonText.contains("4"));
+        assertTrue(totalTestsButtonText.contains("Total Tests"));
+
+        String passedButtonText = getButtonText("PASSED", build);
+        assertTrue(passedButtonText.contains("1"));
+        assertTrue(passedButtonText.contains("Passed"));
+
+        String failedButtonText = getButtonText("FAILED", build);
+        assertTrue(failedButtonText.contains("3"));
+        assertTrue(failedButtonText.contains("Failed"));
+
+        String incompleteButtonText = getButtonText("INCOMPLETE", build);
+        assertTrue(incompleteButtonText.contains("0"));
+        assertTrue(incompleteButtonText.contains("Incomplete"));
+
+        String notRunButtonText = getButtonText("NOT_RUN", build);
+        assertTrue(notRunButtonText.contains("0"));
+        assertTrue(notRunButtonText.contains("Not Run"));
+
+        // Verify the test filenames shown
+        String[] expectedTestFiles = {" testMultiply ", " squareTest ", " testSquare ", " testSum "};
+        String[] testFiles = getTestFilesInTable(build);
+        assertEquals(testFiles, expectedTestFiles);
+
+        // Verify Diagnostics are shown for a failed test.
+        List<String[]> testsContentForAFile = getTestsInfoForATestFile("testMultiply", build);
+        assertEquals(testsContentForAFile.size(),1);
+        assertEquals(testsContentForAFile.get(0)[0].trim(), "testMultiplication"); // 'testMultiply has 'testMultiplication' test
+        assertNotNull(testsContentForAFile.get(0)[1]); // Diagnostics cant be null as 'testMultiply' fails
+        assertTrue(testsContentForAFile.get(0)[1].contains("testMultiply/testMultiplication"));
+        assertNotNull(testsContentForAFile.get(0)[2]); // Duration cannot be null
```

Comment:
Should we also add a test verify if the test case status is correct

Reply:
We are covering this already in unit tests so we can skip that verification at system level